### PR TITLE
Changelogs for -17

### PR DESCRIPTION
**Pull request by martinthomson**
Not sure that I captured everything here.

I also discovered a few things in -recovery that I wasn't paying close attention to that looked bad.  Issues raised for those.

This doesn't include #2045, though I think we agreed to merge it.

### CODE REVIEW

**Comment by janaiyengar** [comment]
Looks good to me.

File: draft-ietf-quic-http.md [hunk 1]
```diff
@@ -1916,13 +1916,17 @@ Error codes need to be defined for HTTP/2 and HTTP/3 separately.  See
...
+  - Remove exclusive reprioritization
+  - Changes to Prioritized Element Type bits
 - Define DUPLICATE_PUSH frame to refer to another PUSH_PROMISE (#2072)
 - Set defaults for settings, allow request before receiving SETTINGS (#1809,
   #1846, #2038)
 - Clarify message processing rules for streams that aren't closed (#1972, #2003)
-- Removed reservation of error code 0 (#1922)
+- Removed reservation of error code 0 and moved HTTP_NO_ERROR to this value
+  (#1922)
+- Removed prohibition of zero-length DATA frames (#2098)
```
Comment (dtikhonov) [inline draft-ietf-quic-http.md]: The change in PR #2098 was reverted.

Comment (janaiyengar) [reply draft-ietf-quic-http.md]: Good catch!

Comment (MikeBishop) [reply draft-ietf-quic-http.md]: I reverted the change in definition of zero, but I kept the un-prohibition of sending zero-length frames.  There seemed to be clear agreement that this was a preference, and there's no technical reason to prohibit it.

Comment (dtikhonov) [reply draft-ietf-quic-http.md]: Oh.  Sorry for the noise.

Comment (janaiyengar) [reply draft-ietf-quic-http.md]: Ah, right. Thanks for reverting my change @MikeBishop :-)

Comment (martinthomson) [reply draft-ietf-quic-http.md]: See, I knew that I'd checked that :)

**Pull request merged**